tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend

```diff
@@ -0,0 +1,441 @@
+package org.testeditor.tcl.dsl.ui.tests.editor
+
+import com.google.common.base.Strings
+import java.nio.file.Files
+import java.nio.file.Paths
+import javax.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.xtext.serializer.ISerializer
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.aml.AmlModel
+import org.testeditor.aml.Component
+import org.testeditor.aml.ComponentElement
+import org.testeditor.dsl.common.testing.AbstractTest
+import org.testeditor.dsl.common.testing.DslParseHelper
+import org.testeditor.tcl.ComponentTestStepContext
+import org.testeditor.tcl.TclModel
+import org.testeditor.tcl.dsl.TclStandaloneSetup
+import org.testeditor.tcl.dsl.ui.editor.DropTargetXtextEditorListener
+import com.google.common.io.ByteStreams
+
+class DropTargetListenerUpdateModelTest extends AbstractTest {
```

Comment:
There are two things I had in mind here:

1. Given / When / Then pattern - see most of the other tests plus http://martinfowler.com/bliki/GivenWhenThen.html
2. The helper methods might be useful elsewhere and/or can be simplified. Having those exclusively in this test will result in a test that is hard to maintain. Imagine if we change the internal model of AML / TCL and have this kind of code in every test-case. We would need to adapt each one of them.